Attendance/Projects/Presentation/Presentation/Sources/CoreMember/QrCode/View/QrCodeView.swift

```diff
@@ -13,151 +13,151 @@ import CoreImage.CIFilterBuiltins
+  var backAction: () -> Void
+  
+  init(
+    store: StoreOf<QrCode>,
+    backAction: @escaping () -> Void
+  ) {
+    self.store = store
+    self.backAction = backAction
+  }
+  
   public  var body: some View {
-      ZStack {
-          Color.basicBlack
-              .edgesIgnoringSafeArea(.all)
-          
-          VStack {
-              Spacer()
-                  .frame(height: 16)
-              
-              NavigationBackButton(buttonAction: backAction)
-              
-              if store.loadingQRImage == true {
-                  TooltipShape(tooltipText: store.qrCodeReaderText)
-                      .offset(y: UIScreen.screenHeight * 0.2)
-              } else {
-                  TooltipShape(tooltipText: store.qrCodeReaderText)
-                      .offset(y: UIScreen.screenHeight * 0.2)
-              }
-              
-              Spacer()
-                  .frame(height: 24)
-              
-//              generateQrImage()
-                  
-              creatEventButton()
-              
-              Spacer()
-          }
-          .navigationBarBackButtonHidden()
-          .task {
-              store.send(.view(.appearLoading))
-              store.send(.async(.fetchEvent))
-              store.send(.async(.observeEvent))
-              
-              Task {
-
-                  store.send(.async(.generateQRCode))
-                  
-              }
-          }
-      }
-      .onChange(of: store.eventModel) { oldValue , newValue in
-          store.send(.async(.updateEventModel(newValue)))
-      }
+    ZStack {
+      Color.basicBlack
+        .edgesIgnoringSafeArea(.all)
       
-      .sheet(item: $store.scope(state: \.destination?.makeEvent, action: \.destination.makeEvent)) { makeEventStore in
-          MakeEventView(store: makeEventStore, completion: {
-              store.send(.view(.closeMakeEventModal))
-          })
-          .presentationDetents([.height(UIScreen.screenHeight * 0.65)])
-          .presentationCornerRadius(20)
-          .presentationDragIndicator(.hidden)
+      VStack {
+        Spacer()
+          .frame(height: 16)
+        
+        NavigationBackButton(buttonAction: backAction)
+        
+        if store.loadingQRImage == true {
+          TooltipShape(tooltipText: store.qrCodeReaderText)
+            .offset(y: UIScreen.screenHeight * 0.2)
+        } else {
+          TooltipShape(tooltipText: store.qrCodeReaderText)
+            .offset(y: UIScreen.screenHeight * 0.2)
+        }
+        
+        Spacer()
+          .frame(height: 24)
+        
+        //              generateQrImage()
+        
+        creatEventButton()
+        
+        Spacer()
+      }
+      .navigationBarBackButtonHidden()
+      .task {
+        store.send(.view(.appearLoading))
+        store.send(.async(.fetchEvent))
+        store.send(.async(.observeEvent))
+        
+        Task {
+          
+          store.send(.async(.generateQRCode))
+          
+        }
       }
```

Comment:
```suggestion
      .task {
        store.send(.view(.appearLoading))
        store.send(.async(.fetchEvent))
        store.send(.async(.observeEvent))
        store.send(.async(.generateQRCode))  
      }
```

`.task` 내부에 중복으로 Task가 있어서 제거하는게 좋을 것 같아요!
더 개선하자면 따로 store에서 onAppear와 같은 action을 따로 추가하고 그안에서 로직을 처리하는게 더 좋아보입니다!

Reply:
@Roy-wonji  수정은 했습니다